### PR TITLE
[core] [9/N] Skip uv install if match

### PR DESCRIPTION
**Pull request by dentiny**
Followup on https://github.com/ray-project/ray/pull/48634, which skip installation if specified version matches what we have in virtual env.

### CODE REVIEW

**Comment by jjyao** [comment]
I think we can always reinstall to make the code simpler which should be an no-op by uv?

**Comment by dentiny** [comment]
> I think we can always reinstall to make the code simpler which should be an no-op by uv?

If user requested version is always installed on the env, why bother re-installing?

**Comment by dentiny** [comment]
> I think we can always reinstall to make the code simpler which should be an no-op by uv?

The win for this PR is to reduce runtime overhead by `pip`, which is known to be ineffective.
I would like to reduce runtime setup invocation via `pip` as much as possible.

File: python/ray/_private/runtime_env/uv.py [hunk 1]
```diff
@@ -117,11 +119,42 @@ async def _check_uv_existence(
...
+            version_output = await check_output_cmd(
+                check_existence_cmd, logger=logger, cwd=cwd, env=env
+            )
+
+            # If exists, the output format would look like
+            # uv <version> (<sha> <release date>), for example,
+            # uv 0.5.1 (f399a5271 2024-11-08)
+            version_strs = version_output.split()
+            if len(version_strs) == 4 and version_strs[0] == "uv":
+                return version_strs[1]
```
Comment (jjyao) [inline python/ray/_private/runtime_env/uv.py:131]: I think this is fragile. The output format can change at anytime. Do we have other way to get the version?

**Comment by jjyao** [comment]
> The win for this PR is to reduce runtime overhead by pip, which is known to be ineffective.

I was thinking that pip should do nothing if the specified version is already installed (basically pip does the skip for us)

**Comment by dentiny** [comment]
> I was thinking that pip should do nothing if the specified version is already installed (basically pip does the skip for us)

That makes sense, though I'm thinking to skip `pip` operation directly.

**Comment by dentiny** [comment]
@jjyao I redo the PR just to skip "enforce reinstallation": https://github.com/ray-project/ray/pull/48731

**Pull request closed without merging**